### ISSUE TITLE
ERC1155BalanceOfBatchExample not working correctly

### ISSUE THREAD

**Issue by juans-chainsafe** [ChainSafe/web3.unity#496]
**Describe the bug**
Looks like this script if failing when it tries to parse the value

**To Reproduce**
1. Add a empty GameObject
2. Add the script **ERC1155BalanceOfBatchExample** to the game object
3. Try the script with the login scene

**Expected behavior**
A log with the balance of the batch is displayed

**Screenshots**
<img width="440" alt="Screenshot 2023-06-27 at 13 24 09" src="https://github.com/ChainSafe/web3.unity/assets/104365141/057e593c-db72-45b1-8747-09100b7e4fd0">

**Additional context**
**This was found while testing https://github.com/ChainSafe/web3.unity/pull/486**

```
System.Object[]
UnityEngine.Debug:LogError (object)
Web3Unity.Scripts.Library.ETHEREUEM.EIP.ERC1155/<BalanceOfBatch>d__2:MoveNext () (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/src/UnityPackages/io.chainsafe.web3-unity/Runtime/Scripts/ETHEREUEM/EIP/ERC1155.cs:54)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<object[]>:SetResult (object[])
Web3Unity.Scripts.Library.Ethers.Contracts.Contract/<Call>d__8:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<string>:SetResult (string)
Web3Unity.Scripts.Library.Ethers.Providers.BaseProvider/<Call>d__37:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<string>:SetResult (string)
Web3Unity.Scripts.Library.Ethers.Providers.JsonRpcProvider/<Perform>d__10`1<string>:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<string>:SetResult (string)
Web3Unity.Scripts.Library.Ethers.Providers.JsonRpcProvider/<Send>d__9`1<string>:MoveNext ()
System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1<ChainSafe.GamingWeb3.Environment.NetworkResponse`1<Nethereum.JsonRpc.Client.RpcMessages.RpcResponseMessage>>:SetResult (ChainSafe.GamingWeb3.Environment.NetworkResponse`1<Nethereum.JsonRpc.Client.RpcMessages.RpcResponseMessage>)
ChainSafe.GamingWeb3.Unity.UnityHttpClient/<Post>d__6`2<Nethereum.JsonRpc.Client.RpcMessages.RpcRequestMessage, Nethereum.JsonRpc.Client.RpcMessages.RpcResponseMessage>:MoveNext ()
System.Threading.Tasks.TaskCompletionSource`1<ChainSafe.GamingWeb3.Environment.NetworkResponse`1<string>>:TrySetResult (ChainSafe.GamingWeb3.Environment.NetworkResponse`1<string>)
Web3Unity.Scripts.Library.Ethers.Unity.Dispatcher/<>c__DisplayClass6_0`1/<<EnqueueTask>g__WrappedAction|0>d<ChainSafe.GamingWeb3.Environment.NetworkResponse`1<string>>:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<ChainSafe.GamingWeb3.Environment.NetworkResponse`1<string>>:SetResult (ChainSafe.GamingWeb3.Environment.NetworkResponse`1<string>)
ChainSafe.GamingWeb3.Unity.UnityHttpClient/<>c__DisplayClass4_0/<<PostRaw>b__0>d:MoveNext ()
System.Runtime.CompilerServices.AsyncMethodBuilderCore/MoveNextRunner:Run ()
IEnumeratorAwaitExtensions:RunOnUnityScheduler (System.Action)
IEnumeratorAwaitExtensions/SimpleCoroutineAwaiter`1<UnityEngine.AsyncOperation>:Complete (UnityEngine.AsyncOperation,System.Exception)
IEnumeratorAwaitExtensions/InstructionWrappers/<ReturnSelf>d__2`1<UnityEngine.AsyncOperation>:MoveNext ()
UnityEngine.SetupCoroutine:InvokeMoveNext (System.Collections.IEnumerator,intptr) (at /Users/bokken/build/output/unity/unity/Runtime/Export/Scripting/Coroutines.cs:17)

FormatException: The value could not be parsed.
System.Numerics.BigNumber.ParseBigInteger (System.ReadOnlySpan`1[T] value, System.Globalization.NumberStyles style, System.Globalization.NumberFormatInfo info) (at <0d13c2264bc34dd2bf77d702ce447db4>:0)
System.Numerics.BigNumber.ParseBigInteger (System.String value, System.Globalization.NumberStyles style, System.Globalization.NumberFormatInfo info) (at <0d13c2264bc34dd2bf77d702ce447db4>:0)
System.Numerics.BigInteger.Parse (System.String value, System.Globalization.NumberStyles style, System.IFormatProvider provider) (at <0d13c2264bc34dd2bf77d702ce447db4>:0)
System.Numerics.BigInteger.Parse (System.String value, System.Globalization.NumberStyles style) (at <0d13c2264bc34dd2bf77d702ce447db4>:0)
System.Numerics.BigInteger.Parse (System.String value) (at <0d13c2264bc34dd2bf77d702ce447db4>:0)
Web3Unity.Scripts.Library.ETHEREUEM.EIP.ERC1155+<>c.<BalanceOfBatch>b__2_0 (System.Object x) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/src/UnityPackages/io.chainsafe.web3-unity/Runtime/Scripts/ETHEREUEM/EIP/ERC1155.cs:50)
System.Linq.Enumerable+SelectArrayIterator`2[TSource,TResult].ToList () (at <5e7f3b3c515942079d80178a15a08413>:0)
System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) (at <5e7f3b3c515942079d80178a15a08413>:0)
Web3Unity.Scripts.Library.ETHEREUEM.EIP.ERC1155.BalanceOfBatch (ChainSafe.GamingWeb3.Web3 web3, System.String contractAddress, System.String[] accounts, System.String[] tokenIds) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/src/UnityPackages/io.chainsafe.web3-unity/Runtime/Scripts/ETHEREUEM/EIP/ERC1155.cs:55)
ERC1155BalanceOfBatchExample.Start () (at Assets/Scripts/Samples/ERC1155/ERC1155BalanceOfBatchExample.cs:14)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) (at <b6c5d0f08ddf477186f780f49bb14dee>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at /Users/bokken/build/output/unity/unity/Runtime/Export/Scripting/UnitySynchronizationContext.cs:153)
UnityEngine.UnitySynchronizationContext.Exec () (at /Users/bokken/build/output/unity/unity/Runtime/Export/Scripting/UnitySynchronizationContext.cs:83)
UnityEngine.UnitySynchronizationContext.ExecuteTasks () (at /Users/bokken/build/output/unity/unity/Runtime/Export/Scripting/UnitySynchronizationContext.cs:107)
```

**Comment by Arshia001** [comment]
Fixed in #486

**Comment by juans-chainsafe** [comment]
Working as expected:
<img width="289" alt="Screenshot 2023-07-10 at 13 39 09" src="https://github.com/ChainSafe/web3.unity/assets/104365141/62207411-1a65-4771-804a-9a26acd99174">

@Arshia001 just, I don't know why it prints the log twice on the console, code looks fine:
 ```
 async void Start()
    {
        string contract = "0x3E0C0447e47d49195fbE329265E330643eB42e6f";
        string account = "0xd25b827D92b0fd656A1c829933e9b0b836d5C3e2";

        BigInteger balanceOf = await ERC20.BalanceOf(Web3Accessor.Web3, contract, account);
        Debug.Log("Balance Of: " + balanceOf);
    } 
```

<img width="702" alt="Screenshot 2023-07-10 at 13 40 31" src="https://github.com/ChainSafe/web3.unity/assets/104365141/a52cda03-1e83-4f7e-8ede-624104eabbf0">
